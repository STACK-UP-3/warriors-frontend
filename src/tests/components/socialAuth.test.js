import React from 'react';
import { mount } from 'enzyme';
import SocialAuth from '../../components/socialAuth';

let wrapper,handler;
const { location } = window;

beforeAll(()=>{
    wrapper = mount(<SocialAuth />);
    delete window.location;
    window.location = { assign : jest.fn() }
})

afterAll(() => {
    window.location = location;
});

test('should render correctly',()=>{
    expect(wrapper).toMatchSnapshot();
})
 
test('should simulate click on google',()=>{
    wrapper.find('#google').simulate('click')
    expect(window.location.assign).toHaveBeenCalled()
}) 

test('should simulate click on facebook',()=>{
    wrapper.find('#facebook').simulate('click')
    expect(window.location.assign).toHaveBeenCalled()
})
