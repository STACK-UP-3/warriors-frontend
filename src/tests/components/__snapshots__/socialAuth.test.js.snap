// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<SocialAuth>
  <div
    className="socialAuth"
  >
    <p>
      or login with:
    </p>
    <div
      className="socialAuthImg"
    >
      <div
        id="google"
        onClick={[Function]}
      >
        <img
          src="../assets/google.svg"
        />
      </div>
      <div
        id="facebook"
        onClick={[Function]}
      >
        <img
          src="../assets/facebook.png"
        />
      </div>
    </div>
  </div>
</SocialAuth>
`;
